test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -5158,36 +5239,121 @@ def save_configuration(self, outfile, mode='a'):
+                qmgr, '-c', 'print server'], sudo=True,
+            logerr=False, level=logging.DEBUG)
         if ret['rc'] != 0:
             return False
         else:
             conf['qmgr_print_server'] = ret['out']
-
-        ret = self.du.run_cmd(self.hostname, [qmgr, '-c', 'p sched'],
-                              sudo=True)
+        ret = self.du.run_cmd(self.hostname, [qmgr, '-c', 'print sched'],
+                              logerr=False, level=logging.DEBUG, sudo=True)
         if ret['rc'] != 0:
             return False
         else:
             conf['qmgr_print_sched'] = ret['out']
+        ret = self.du.run_cmd(self.hostname, [pbsnodes, '-av'],
+                              logerr=False, level=logging.DEBUG)
+        nodes_val = self.utils.convert_to_dictlist(ret['out'])
+        conf['pbsnodes'] = nodes_val
+        try:
+            with open(outfile, mode) as f:
+                json.dump(conf, f)
+        except BaseException:
+            self.logger.error('Error processing file ' + outfile)
+            return False
 
-        ret = self.du.run_cmd(self.hostname, [qmgr, '-c', 'p h'], sudo=True)
+        return True
+
+    def _save_hook_files(self):
+        """
+        save all the hooks .CF, .PY, .HK files
+        """
+        qmgr = os.path.join(self.client_conf['PBS_EXEC'], 'bin', 'qmgr')
+        cfg = {"hooks": ""}
+        cmd = [qmgr, '-c', 'print hook @default']
+        ret = self.du.run_cmd(self.hostname, cmd,
+                              sudo=True)
         if ret['rc'] != 0:
             return False
         else:
-            conf['qmgr_print_hook'] = ret['out']
+            cfg['qmgr_print_hook'] = ret['out']
+        return cfg
+
+    def save_configuration(self, outfile, mode='a'):
+        """
+        Save a server configuration, this includes:
+
+          - ``server_priv/resourcedef``
 
+          - ``qmgr -c "print server"``
+
+          - ``qmgr -c "print sched"``
+
+          - ``qmgr -c "print hook"``
+
+        :param outfile: the output file to which onfiguration is
+                        saved
+        :type outfile: str
+        :param mode: The mode in which to open outfile to save
+                     configuration. The first object being saved
+                     should open this file with 'w' and subsequent
+                     calls from other objects should save with
+                     mode 'a' or 'a+'. Defaults to a+
+        :type mode: str
+        :returns: True on success, False on error
+        """
+        conf = {}
+        sconf = {MGR_OBJ_SERVER: conf}
+        # save pbs.conf file
+        cfg_path = self.du.get_pbs_conf_file()
+        with open(cfg_path, 'r') as p:
+            pbs_cfg = p.readlines()
+            config = self.utils.convert_to_dictlist(pbs_cfg)
+            pbs_cfg_b64 = base64.encodestring(str(config[0]))
+        conf['pbs_conf'] = pbs_cfg_b64
+        # save hook files
+        hooks_str = self._save_hook_files()
```

Comment:
I think you should add a check here - if `_save_hook_files()` is returning `False` then you should maybe throw an exception or so?

Reply:
Changed, returning false when saving of hook has failed.